File: 605. Can Place Flowers/solution.go
func canPlaceFlowers(flowerbed []int, n int) bool {
	for i, size := 0, len(flowerbed); i < size; i++ {
		if flowerbed[i] == 0 &&
			((i+1 < size && flowerbed[i+1] == 0) || i+1 >= size) &&
			((i-1 >= 0 && flowerbed[i-1] == 0) || i-1 < 0) {
			flowerbed[i] = 1
			n--

			if n <= 0 {
				return true
			}
		}
	}

	return n <= 0
}

File: 605. Can Place Flowers/solution.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
  let i = 0;

  while (i < flowerbed.length && n > 0) {
    if (~~flowerbed[i - 1] + ~~flowerbed[i + 1] + flowerbed[i] === 0) {
      n--;
      i++;
    }

    i++;
  }

  return n === 0;
};
